### PR TITLE
NuGet publish

### LINKED ISSUE #38: Fix NuGet publishing
I have no idea why, but neither Travis nor AppVeyor is running our builds, so even though #37 is merged, tagged and released, there won't be an automatic push of the `.nupkg` to NuGet.org. We should just delete `appveyor.yml` and `.travis.yml` and set up auto-publishing to NuGet.org with GitHub Actions instead.

### PR DESCRIPTION
**Pull request by AlexanderStromer**
1. Manual Trigger of workflow (see Actions -> Nuget Publish)
2. All ready to go, **but** before you try to publish you need to setup the APIKey secret so you can publish to nuget:
  2a: In Nuget.org -> your name (top right) > API Keys -> Create new key and copy
  2b: In GitHub -> NHtmlUnit -> Settings -> Secrets (only visible to Contributors) -> Add a new secret called "NUGET_APIKEY" with the APIKey as value

Resolves #38.

### CODE REVIEW

**Comment by AlexanderStromer** [comment]
wrt #38

File: .github/workflows/nugetpublish.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      run: nuget restore NHtmlUnit.sln
+      
+    - name: Build Solution
+      run: msbuild NHtmlUnit.sln /property:Configuration=Release
+      
+    - name: Create NuGet Package
+      run: nuget pack NHtmlUnit.nuspec
+
+    - name: Publish To NuGet.org
+      run: nuget push *.nupkg -Source nuget.org -ApiKey ${{ secrets.NuGet_APIKey }}
```
Comment (asbjornu) [inline .github/workflows/nugetpublish.yml:29]:
```suggestion
      run: nuget push *.nupkg -Source nuget.org -ApiKey ${{ secrets.NUGET_API_KEY }}
```

**Review by asbjornu — CHANGES_REQUESTED** [review]
Brilliant!

File: .github/workflows/nugetpublish.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: Nuget Publish
+
+on: workflow_dispatch
```
Comment (asbjornu) [inline .github/workflows/nugetpublish.yml:3]: It would be great to release to NuGet when a release (and tag) is created on GitHub, so the only manual labour required would be to create the GitHub release. From it we also get the version number we can use for the NuGet package (so it no longer needs to be hard coded). [GitVersion](https://github.com/marketplace/actions/gittools) can assist in producing a version number from the created tag.

**Review by asbjornu — CHANGES_REQUESTED** [review]

File: .github/workflows/nugetpublish.yml (hunk 2 above)
Comment (AlexanderStromer) [reply .github/workflows/nugetpublish.yml:3]: I thought about this and had a look at GitVersion. I think you already did a good job adopting the version of HtmlUnit for NHtmlUnit (assuming NHtmlUnit does not have its own releases as such, e.g. for fixes). Also, adopting this auto-versioning-workflow makes it a mixed bag of what is actually getting published/updated automatically. I'd prefer a simple straightforward approach, which is either
a. **all** version handling is done by GitHub workflows
b. **no** version handling is done by GitHub workflows
Since NHtmlUnit is highly dependent on HtmlUnit's version, which requires manual handling anyway, I would recommend option b
Simply put, given the mixed-bag outcome and effort to implement GitVersion, I don't think its worth it.

As for NuGet publish when making a GitHub release, this is easy to do, you can add similar as seen here >
[SharpZipLib Using Release](https://github.com/icsharpcode/SharpZipLib/blob/master/.github/workflows/release.yml)

Comment (asbjornu) [reply .github/workflows/nugetpublish.yml:3]: I'm not saying the version number should be minted automatically. I'm saying the version number manually minted when creating the release should be the version number stamped in the published `.nupkg` so we don't have to manually keep two version numbers in tight synchronization. Thinking about it, GitVersion isn't necessary since the we can just extract the version number directly from `github.ref`.

Comment (AlexanderStromer) [reply .github/workflows/nugetpublish.yml:3]: Looks like you know more about this than me :) I'll leave that for you.

**Comment by asbjornu** [comment]
Thanks for moving this in the right direction, @AlexanderStromer! #40 and a few more PRs are now merged and [version 2.50.0 is now published on NuGet.org](https://www.nuget.org/packages/NHtmlUnit/2.50.0). 🎉
